src/backend/tests/Fusion.Testing.Mocks.ProfileService/PeopleServiceMock.cs

```diff
@@ -58,9 +62,59 @@ public FusionTestUserBuilder WithPreferredContactMail(string mail)
             return this;
         }
 
+        /// <summary>
+        /// Will flag the profile as resource owner by setting the bool property to true. 
+        /// The profile will also be added to the department above or CEC.
+        /// 
+        /// A LineOrg role will be added as well.
+        /// 
+        /// Should be called after the full department has been updated.
+        /// </summary>
+        /// <returns></returns>
         public FusionTestUserBuilder AsResourceOwner()
         {
             profile.IsResourceOwner = true;
+
+            try
+            {
+                var depTokens = profile.FullDepartment.Split(' ');
+                // Lift manager 1 level up. 
+
+                var parentFullDepartment = string.Join(" ", depTokens.SkipLast(1));
+                var parentDepartment = string.Join(" ", depTokens.SkipLast(1).TakeLast(3));
+
+                if (string.IsNullOrEmpty(parentDepartment))
+                    parentDepartment = "CEC";
+                if (string.IsNullOrEmpty(parentFullDepartment))
+                    parentFullDepartment = "CEC";
+                
+                profile.FullDepartment = parentFullDepartment;
+                profile.Department = parentDepartment;
+
+            }
+            catch (Exception) {  /* */ }
```

Comment:
It is because there might not be set a department.. which then it will fail.. It should be discovered when validating outcome that departments are not set up correctly. Too much time to make this 100% safe. As you pointed out, it is just mock.